controller/index.js

```diff
@@ -22,16 +22,30 @@ var util = require('util'),
     path = require('path'),
     yeoman = require('yeoman-generator'),
     krakenutil = require('../util'),
-    prompts = require('./prompts');
-
+    prompts = require('./prompts'),
+    fs = require('fs');
 
 
 var Generator = module.exports = function Generator(args, options, config) {
     yeoman.generators.Base.apply(this, arguments);
+    var cwd = this.env.cwd,
+        filePath = path.join(cwd, 'package.json'),
+        packageTemplate = null;
 
     krakenutil.update();
 
-    this.hasTemplates = (args[1]) ? true : false;
+    this.hasTemplates = (args[1]) ? true : (
+        fs.existsSync(filePath) ? (
+            function() {
+                packageTemplate = require(filePath.replace('.json', ''))['generator-kraken'].template;
+                return true;
+            })() : false
+    );
```

Comment:
Yeah, that's what it does. It's when it's running in the context of an existing generated project.

From womp's package.json: 

```
  "generator-kraken": {
    "version": "1.1.0",
    "template": "dustjs",
    "css": "less",
    "js": "requirejs",
    "task": "grunt"
  }
```